tests/ui/mismatched_types/issue-118510.rs

```diff
@@ -0,0 +1,13 @@
+pub enum Sexpr<'a> {
+    Ident(&'a mut S),
+    //~^ ERROR cannot find type `S` in this scope [E0412]
+}
+
+fn map<'a, F: FnOnce(&Foo<'a>) -> T>(f: F) {}
+//~^ ERROR cannot find type `Foo` in this scope [E0412]
+//~| ERROR cannot find type `T` in this scope [E0412]
```

Comment:
Could you define `S`, `Foo` and `T`?
All the "cannot find type" errors are off-topic for this test.